src/sys/resource.rs

```diff
@@ -0,0 +1,230 @@
+                RLIMIT_RSS,
+
+                #[cfg(any(target_os = "android", target_os = "linux"))]
+                RLIMIT_RTPRIO,
+
+                #[cfg(any(target_os = "linux"))]
+                RLIMIT_RTTIME,
+
+                #[cfg(any(target_os = "android", target_os = "linux"))]
+                RLIMIT_SIGPENDING,
+
+                #[cfg(any(target_os = "freebsd", target_os = "dragonfly"))]
+                RLIMIT_SBSIZE,
+
+                #[cfg(target_os = "freebsd")]
+                RLIMIT_SWAP,
+
+                #[cfg(target_os = "freebsd")]
+                RLIMIT_VMEM,
+            }
+        }
+    }else{
+        // unkown os
+    }
+}
+
+/// Get the current processes resource limits
+///
+/// A value of `None` indicates that there's no limit.
+///
+/// # Parameters
+///
+/// * `resource`: The [`Resource`] that we want to get the limits of.
+///
+/// # Examples
+///
+/// ```
+/// # use nix::sys::resource::{getrlimit, Resource};
+///
+/// let (soft_limit, hard_limit) = getrlimit(Resource::RLIMIT_NOFILE).unwrap();
+/// println!("current soft_limit: {:?}", soft_limit);
+/// println!("current hard_limit: {:?}", hard_limit);
+/// ```
+///
+/// # References
+///
+/// [getrlimit(2)](https://linux.die.net/man/2/getrlimit)
+///
+/// [`Resource`]: enum.Resource.html
+pub fn getrlimit(resource: Resource) -> Result<(Option<rlim_t>, Option<rlim_t>)> {
+    let mut old_rlim = rlimit {
+        rlim_cur: 0,
+        rlim_max: 0,
+    };
+
+    cfg_if! {
+        if #[cfg(all(target_os = "linux", target_env = "gnu"))]{
+            let res = unsafe { libc::getrlimit(resource as __rlimit_resource_t, &mut old_rlim) };
+        }else if #[cfg(any(
+            target_os = "freebsd",
+            target_os = "openbsd",
+            target_os = "netbsd",
+            target_os = "macos",
+            target_os = "ios",
+            target_os = "android",
+            target_os = "dragonfly",
+            target_os = "bitrig",
+            target_os = "linux", // target_env != "gnu"
+        ))]{
+            let res = unsafe { libc::getrlimit(resource as c_int, &mut old_rlim) };
+        }
+    }
+
+    Errno::result(res).map(|_| {
+        (
+            Some(old_rlim.rlim_cur).filter(|x| *x != RLIM_INFINITY),
+            Some(old_rlim.rlim_max).filter(|x| *x != RLIM_INFINITY),
+        )
+    })
+}
+
+/// Set the current processes resource limits
+///
+/// A value of `None` indicates that there's no limit.
+///
+/// # Parameters
+///
+/// * `resource`: The [`Resource`] that we want to set the limits of.
+/// * `soft_limit`: The value that the kernel enforces for the corresponding
+///   resource. Note: `None` input will be replaced by constant `RLIM_INFINITY`.
+/// * `hard_limit`: The ceiling for the soft limit. Must be lower or equal to
+///   the current hard limit for non-root users. Note: `None` input will be
+///   replaced by constant `RLIM_INFINITY`.
+///
+/// # Examples
+///
+/// ```no_run
```

Comment:
Why `no_run`?  Doc tests each get their own process, so this shouldn't interfere with other tests.

Reply:
That's what was written in previous PR. I will remove this `no_run`